rocky/reports/views/base.py

```diff
@@ -169,26 +169,130 @@ def get_report_types_from_choice(self) -> list[type[Report] | type[MultiReport]]
                 messages.add_message(self.request, messages.ERROR, error_message)
         return report_types
 
-    def get_report_types(self) -> list[ReportType]:
+    def get_report_types(self, reports: set[type[BaseReportType]]) -> list[dict[str, str]]:
         return [
             {
                 "id": report_type.id,
                 "name": report_type.name,
                 "description": report_type.description,
                 "label_style": report_type.label_style,
             }
-            for report_type in self.get_report_types_from_choice()
+            for report_type in reports
         ]
 
+
+class ReportPluginView(ReportOOIView, ReportTypeView, TemplateView):
+    """
+    This view shows the required and optional plugins.
+    Needs ReportTypeView to know which report type is selected to get their plugins.
+    The plugin ids will be collected and fetched form KAT-alogus.
+    The user is able to activate plugins they need for the scans.
+    The oois selection is also remembered by ReportOOIView.
+    """
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        try:
+            self.plugins, self.all_plugins_enabled = self.get_plugins()
+        except httpx.HTTPStatusError:
+            messages.error(
+                self.request,
+                _(
+                    "One or more plugins of the selected reports does not exist, "
+                    "therefore the report cannot be generated."
+                ),
+            )
+            self.plugins = {}
+            self.all_plugins_enabled = {}
+
+    def get(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        if not self.selected_report_types:
+            messages.error(request, _("Select at least one report type to proceed."))
+        return super().get(request, *args, **kwargs)
+
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        context["created_at"] = datetime.now()
-        context["selected_oois"] = self.selected_oois
-        context["selected_report_types"] = self.selected_report_types
+
         context["plugins"] = self.plugins
-        context["oois"] = self.get_oois()
+        context["all_plugins_enabled"] = self.all_plugins_enabled
+        context["plugin_data"] = self.get_plugin_data()
+
         return context
 
+    def plugins_enabled(self) -> bool:
+        if self.all_plugins_enabled and self.all_plugins_enabled:
+            return self.all_plugins_enabled["required"] and self.all_plugins_enabled["optional"]
```

Comment:
Oww this is a typo I see here.

should be this only:
`return self.all_plugins_enabled["required"] and self.all_plugins_enabled["optional"]`

Reply:
Why do we need to keep the `if` here? Where do we set `all_plugins_enabled` to something that would evaluate to `False`?

Reply:
Looking further at the code I see that it can happen when the Katalogus returns an error. I think we should just display an error page instead of trying to continu. Continuing means we have to add if statements in those methods that only check if there is an error and then return wrong data because we couldn't fetch the data.

But let's refactor this later so this PR can be merged now.